modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotRestoreProcess.java

```diff
@@ -1842,10 +1846,18 @@ private static void completeListExceptionally(List<PartitionRestoreFuture> col,
      * @param map Map of partitions and cache groups.
      * @return String representation.
      */
-    private static String partitionsMapToString(Map<Integer, Set<Integer>> map) {
+    private String partitionsMapToString(Map<Integer, Set<Integer>> map) {
+        Map<Integer, String> cacheGrpNames = new HashMap<>(opCtx.dirs.size());
+
+        opCtx.dirs.forEach(dir -> {
+            String grpName = cacheGroupName(dir);
+            cacheGrpNames.put(GridCacheUtils.cacheId(grpName), grpName);
+        });
+
         return map.entrySet()
             .stream()
-            .collect(Collectors.toMap(Map.Entry::getKey, e -> S.toStringSortedDistinct(e.getValue())))
+            .collect(Collectors.toMap(e -> String.format("[grpId=%d, grpName=%s]", e.getKey(), cacheGrpNames.get(e.getKey())),
+                e -> GridToStringBuilder.toStringSortedDistinct(e.getValue())))
```

Comment:
Is it necessary to revert from `S`?

Reply:
I've done this to avoid errors stated by Sonar, I will try to leave as it was previously if it is possible